src/Language/PureScript/Make.hs

```diff
@@ -171,21 +90,27 @@ make ma@MakeActions{..} ms = do
 
   (sorted, graph) <- sortModules ms
 
-  barriers <- zip (map getModuleName sorted) <$> replicateM (length ms) ((,) <$> C.newEmptyMVar <*> C.newEmptyMVar)
```

Comment:
The real change in this PR is this section, which is now broken down between `Make.BuildPlan` and here.

Instead of checking the timestamps of the dependencies of a module before building it, we now check all the timestamps once, and make a plan of which modules need to be rebuilt (`BuildPlan.construct`).